pkg/apply/merge_test.go

```diff
@@ -107,6 +107,38 @@ metadata:
 	}))
 }
 
+func TestMergeOne(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	cur := UnstructuredFromYaml(t, `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: d1
+  labels:
+    c: upd
+  annotations:
+    c: upd`)
```

Comment:
nit: took me a while to get this test. can you change it in:
```suggestion
  labels:
    label-c: cur
  annotations:
    annotation-c: cur`)
```